support/tests/test_tealium_debug/TealiumDebugServerTests.swift

```diff
@@ -49,6 +50,28 @@ class TealiumDebugServerTests: XCTestCase {
         
     }
     
+    func testQueueMaxLimit () {
+      
+        let debugServer = TealiumDebugServer()
+        debugServer.setQueueMax(50)
+        
+        for i in 1..<53 {
+            debugServer.add(["\(i)": "\(i)"])
+        }
+      
+        XCTAssertTrue(debugServer.debugQueue.count == debugServer.queueMax, " expected count of  \(debugServer.queueMax) does not match queue count:  \(debugServer.debugQueue.count)")
+        
+        //first item should be removed
+        if (debugServer.debugQueue.contains { $0 == ["1":"1"] } == true) {
```

Comment:
I don't believe this is correct:
a) this only checks to see if that entry exists, not that it's in the first position
b) the FIFO array should have values between 3-52 in a 50 item array after 53 adds.